webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/CustomPageJs.scala

```diff
@@ -0,0 +1,27 @@
+package com.normation.rudder.web.snippet
+
+import net.liftweb.http.DispatchSnippet
+import net.liftweb.http.LiftRules
+import net.liftweb.http.S
+
+/**
+  * We need to provide both lift.js and page js to the client since the scripts appended by lift is hard to intercept.
+  * TODO: find a way to remove these duplicates after merging, or parse the whole html and add nonces to the scripts after merging
+  * , see lift source code : 
+  * https://github.com/lift/framework/blob/master/web/webkit/src/main/scala/net/liftweb/http/LiftSession.scala#L995
+  * 
+  * For now both scripts will be duplicated in the page, it does not induce weird behavior
+  */
+class CustomPageJs extends DispatchSnippet {
+
+  def dispatch: DispatchIt = { case "pageScript" => _ => pageScript }
+
+  def pageScript = {
+    <script data-lift="with-nonce" src="/classpath/lift.js"></script>
+    <script type="text/javascript" data-lift="with-nonce" src={scriptUrl(s"page/${S.renderVersion}.js")}></script>
+  }
+
+  private def scriptUrl(scriptFile: String) = {
+    S.encodeURL(s"/${LiftRules.liftContextRelativePath}/$scriptFile")
+  }
```

Comment:
Lifts adds a `<script src="/rudder/classpath/lift.js"/>` and another page script every time an html page is rendered, but it does not currently provide an easy way in the rendering lifecycle to replace/add the attributes to these elements. 

So for now we have console errors but at least each of the scripts are loaded once, and blocked once.

The only level at which it is possible to do it is when generating a `LiftResponse` i.e. with the whole response xml content already generated, and replace those scripts to add the nonce to them... 
I don't know which solution is "cleaner" though

Reply:
OK, we will need to correct that duing alpha (it means: "it's a new task in the milestone `ready for beta`)
Perhaps just have a hook in `LftRules.beforeSend` (or at the last moment before request is still xml, so that we can easily add nonce to script elements, see https://exploring.liftweb.net/master/index-9.html#sec:Request/Response-Lifecycle for the request lifecycle).

Reply:
That's funny in the documentation they mean something different to the source code : 
```
  /**
   * Holds user functions that are executed before sending the response to client. The functions'
   * result will be ignored.
   */
  val beforeSend = RulesSeq[(BasicResponse, HTTPResponse, List[(String, String)], Box[Req]) => Any]
```
VS
```
Call any functions defined in LiftRules.beforeSend. This is the last place where you can modify the response before it’s sent to the user 
```
I'll try that, but since our `LiftResponse` is a `case class` I hope HTTPResponse can be modified easily !